src/mlpack/methods/ann/layer/reparametrization_impl.hpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file reparametrization_impl.hpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Reparametrization class which samples from parameters for a given
+ * distribution.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_REPARAMETRIZATION_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_REPARAMETRIZATION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "reparametrization.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+Reparametrization<InputDataType, OutputDataType>::Reparametrization()
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+Reparametrization<InputDataType, OutputDataType>::Reparametrization(
+    const size_t latentSize) :
+    latentSize(latentSize)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void Reparametrization<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, arma::Mat<eT>&& output)
+{
+  if (input.n_rows != 2 * latentSize)
+  {
+    Log::Fatal << "The output size of layer before the Reparametrization "
+        << "layer should be 2 * latent size of the Reparametrization layer!"
+        << std::endl;
+  }
+
+  mean = input.submat(latentSize, 0, 2 * latentSize - 1, input.n_cols - 1);
+  SoftplusFunction::Fn(input.submat(0, 0, latentSize - 1, input.n_cols - 1),
```

Comment:
we need to store the second vector as well

stddev = input.submat(...)

Reply:
Okay, so there has been some confusion. I am not storing it here because we don't consider that the standard deviation. The first part of the input vector after applying softplus gives the standard deviation, because the input can have negative values.